forms-flow-web/src/components/Form/EditForm/FormEdit.js

```diff
@@ -349,7 +350,12 @@ const EditComponent = () => {
       dispatch(setRestoreFormId(null));
     };
   }, [restoredFormId]);
-
+  useEffect(() => {
+    // Automatically add hidden application components to the form on render/update
+    const updatedForm = addHiddenApplicationComponent(form);
+    dispatchFormAction({ type: "replaceForm", value: updatedForm });
```

Comment:
is some one restored the old version of the form  then need to add  this component right ? @arun-s-aot @abilpraju-aot

Reply:
@shuhaib-aot , what do you mean by restored, import an old form or getting back in history?

Reply:
it may repeat in all renders.